infor-design/enterprise-ng
# WIP: Add ng-add Schematics

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
Adds an initial schematic for `ng add` such that `ng add ids-enterprise-ng` will modify project files according to the QUICKSTART guide:

1. Add scripts to angular.json
2. Add assets to angular.json
3. Add types to tsconfig
4. Add stylesheet link to index.html
5. Import SohoComponentsModule

**Related github/jira issue (required)**:
#793 

**Steps necessary to review your pull request (required)**:
Testing locally:
1. Build & pack: `npm run build:lib && npm run build:schematics && npm run pack:lib`
2. Open another existing project, or create one with `ng new`
3. In the new project, run `ng add ../path/to/enterprise-ng/dist/ids-enterprise-ng/ids-enterprise-ng-x-y-z.tgz`. It may also be possible to use `npm link` for this.
4. Project should now be ready to use the IDS directives/components

**Note:** I'm not sure which build scripts are usually run when building/testing/packaging. This may need to be changed in package.json

**TODOS:**
- [ ] `Soho.Locale.set` on APP_INITIALIZER
- [ ] Add SVG:s to app.component?
- [ ] Add option for uplift/dark/light theme
- [ ] Add option for locale
- [ ] Schematics unit tests
- [ ] Add a README / MD doc about it all
